src/ComplexSelector.js

```diff
@@ -7,31 +7,31 @@ export default class ComplexSelector {
     this.childrenOfNode = childrenOfNode;
   }
 
-  getSelectors(selector) {
-    const selectors = split(selector, / (?=(?:(?:[^"]*"){2})*[^"]*$)/);
-    return selectors.reduce((list, sel) => {
-      if (sel === '+' || sel === '~') {
+  getSelectors(selectorNodes) {
+    return selectorNodes.reduce((list, node) => {
+      if (node.type === 'combinator' && (node.value === '+' || node.value === '~')) {
         const temp = list.pop();
-        list.push(sel, temp);
+        list.push(node, temp);
         return list;
+      } else if (node.type !== 'combinator' || node.value !== ' ') {
+        list.push(node);
       }
 
-      list.push(sel);
       return list;
     }, []);
   }
 
   handleSelectors(selectors, wrapper) {
     const recurseSelector = (offset, fn, pre) => {
-      const predicate = pre || this.buildPredicate(selectors[offset]);
+      const predicate = pre || this.buildPredicate(selectors[offset].toString());
```

Comment:
`toString` is not guaranteed to return a String - please wrap in `String()` instead, throughout.